### ISSUE TITLE
Unable to create db-connected REST services for tables that have an "_" in their name.

### ISSUE THREAD

**Issue by jeremygiberson** [zfcampus/zf-apigility-admin#295]
General case is that when you attempt to create a db-connected REST service for a table that has an underscore in the name the tooling partially errors out during save. If you reload the page you'll see your new service but when you click on the service it is unable to load configurations for that service.

In my particular case I'm configuring Oauth authentication for my API. One of the first REST services I'm adding is a user registration service (doesn't require auth) so that I can create new users via the API. I followed the instructions on https://apigility.org/documentation/modules/zf-oauth2 to setup my sqlite database.

As I walk through the wizard to create my new service everything looks good. It shows my the adapters, it shows the tables in the selected adapter, and shows me details about that table when I expand it. 

However, when I click save the page displays an error message "Unable to fetch service". Examining the network request, I can see these request details:

```
Remote Address:127.0.0.1:8080
Request URL:http://localhost:8080/apigility/api/module/Auth/rest/Auth-V1-Rest-Oauth_users-Controller
Request Method:GET
Status Code:404 Not Found
```

**Comment by jeremygiberson** [comment]
FYI my current work around, for anyone else experiencing this problem, is to create a junk table without an underscore in its name and base my service on that. Afterwards I can go to the database settings for that service and change the table name back to my intended target, drop the junk table. Then the service and management of that service works fine. Which leads me to believe the issue is primarily with the controller naming convention.

**Comment by aidygus** [comment]
Also having the same problem here.  If I create an endpoint against a table with underscores, I can't edit the details with "Error saving field" and I can't delete it  with the message "Unable to delete service".

I'll have a go at your work around jeremy.

**Comment by leonardo-piseri** [comment]
I think this issue is caused by the function capitalizeFirstLetter() in the package zf-apigility-admin-ui. 
This function is applying a different strategy from the function used here in zf-apigility-admin. Here the code is using the filter "WordUnderscoreToCamelCase", in zf-apigility-admin-ui, just the first character of the string is set to upper case. 
I have created a pull request that solve this issue in the package zf-apigility-admin-ui.

**Comment by pradithahidayat** [comment]
I also have the same problem. I've tried to create a database service based with the table name is tbl_user. The service was successfully created, but there is an error like "Error Saving Field".
After that I click the service and then get an error "Unable to fetch service".
I tried to delete the service, but it show the error again "Unable to delete service".

But if I try to create with the table name without underscore character (e.g. tblUser), it's created successfully and there is no error for this service.

Is there anything need to be changed..?

**Comment by leonardo-piseri** [comment]
Hi Preditha, 
I think that this issueis caused by a bug in the zf-apigility-admin-ui module. I have created a pull request on that module that solve this issue, is just a one line change, but it hasn't been accepted yet. 
Here's my pull request: https://github.com/zfcampus/zf-apigility-admin-ui/pull/59
Please let me know if this works also for you. Note you have to use "grunt" to rebuild the code after you apply the change

**Comment by pradithahidayat** [comment]
Hi leonardo,
thanks for your solutions, but could you please help me how to rebuild the code..?
I've already installed the grunt, but I don't know what should I write in the Gruntfile.js

**Comment by leonardo-piseri** [comment]
Hi Praditha,

just refer to the docs you find here
https://github.com/zfcampus/zf-apigility-admin-ui.

If you have all requirements I think you just have to use the command:

$ grunt clean && grunt build







On Mon, Jun 15, 2015 at 12:26 PM, Praditha Hidayat <notifications@github.com

**Comment by weierophinney** [comment]
Fixed with zfcampus/zf-apigility-admin-ui#59, which we'll release in a new version within the next 7 days.

**Comment by tushariscoolster** [comment]
@weierophinney @leonardo-piseri @pradithahidayat  @aidygus @jeremygiberson @jeremiahsmall https://github.com/tushariscoolster/apigility-db-connected-fix

**Comment by colinkelly** [comment]
Like @weierophinney has stated, the fix was added zfcampus/zf-apigility-admin-ui#59 though it got overwritten. I have made a new pull request to add this fix again zfcampus/zf-apigility-admin-ui#100

This is not an issue with the zfcampus/zf-apigility-admin but with the UI. We should keep this issue closed.
